var/spack/repos/builtin/packages/sollve/package.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Sollve(CMakePackage):
+    """The SOLLVE Project aims at scaling OpenMP by leveraging LLVM for exascale
+       performance and portability of applications.  This package provides a
+       collection of Clang/LLVM compilers and an OpenMP runtime library.
+    """
+
+    homepage = 'https://www.bnl.gov/compsci/projects/SOLLVE/'
+    url      = "https://github.com/SOLLVE"
+
+    family = 'compiler'  # Used by lmod
+
+    # NOTE: The debug version of LLVM is an order of magnitude larger than
+    # the release version, and may take up 20-30 GB of space. If you want
+    # to save space, build with `build_type=Release`.
+
+    variant('clang', default=True,
+            description="Build the LLVM C/C++/Objective-C compiler frontend")
+    variant('lldb', default=True, description="Build the LLVM debugger")
+    variant('lld', default=True, description="Build the LLVM linker")
+    variant('internal_unwind', default=True,
+            description="Build the libcxxabi libunwind")
+    variant('polly', default=True,
+            description="Build the LLVM polyhedral optimization plugin, "
+            "only builds for 3.7.0+")
+    variant('libcxx', default=True,
+            description="Build the LLVM C++ standard library")
+    variant('compiler-rt', default=True,
+            description="Build LLVM compiler runtime, including sanitizers")
+    variant('gold', default=True,
+            description="Add support for LTO with the gold linker plugin")
+    variant('shared_libs', default=False,
+            description="Build all components as shared libraries, faster, "
+            "less memory to build, less stable")
+    variant('link_dylib', default=False,
+            description="Build and link the libLLVM shared library rather "
+            "than static")
+    variant('all_targets', default=False,
+            description="Build all supported targets, default targets "
+            "<current arch>,NVPTX,AMDGPU,CppBackend")
+    variant('build_type', default='Release',
+            description='CMake build type',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
+    variant('omp_tsan', default=False,
+            description="Build with OpenMP capable thread sanitizer")
+    variant('python', default=False, description="Install python bindings")
+    variant('argobots', default=True, description="Use Argobots in BOLT")
+    extends('python', when='+python')
+
+    # Build dependency
+    depends_on('cmake@3.4.3:', type='build')
+    depends_on('python', when='~python', type='build')
```

Comment:
```suggestion
    depends_on('python', type='build')
```
It always depends on Python at build time, but it also depends on Python at run-time when `+python`. This should be handled by the `extends()` above.

Reply:
I removed `when` as suggested.